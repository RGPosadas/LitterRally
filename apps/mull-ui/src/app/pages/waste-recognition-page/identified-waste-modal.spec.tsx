import { fireEvent, getByTestId, render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import IdentifiedWasteModal from './identified-waste-modal';

describe('IdentifiedWasteModal', () => {
  it('should render successfully', async () => {
    const { baseElement } = render(<IdentifiedWasteModal />);
    expect(baseElement).toBeTruthy();
  });

  it('should open modal succesfully', async () => {
    const { container } = render(<IdentifiedWasteModal />);
    const button = getByTestId(container, 'open-modal');
    fireEvent.click(button);
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<IdentifiedWasteModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
