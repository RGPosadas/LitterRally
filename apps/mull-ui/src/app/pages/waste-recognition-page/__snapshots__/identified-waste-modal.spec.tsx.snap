// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdentifiedWasteModal should match snapshot 1`] = `
<div
  className="identified-waste-container"
>
  <div
    data-testid="open-modal"
    onClick={[Function]}
  >
    <button>
      CLICK HERE
    </button>
  </div>
</div>
`;
